language/test/Language_test.rb

```diff
@@ -162,6 +164,86 @@
     end
   end
 
+  describe "#Save" do
+    let(:language) { "es_ES" }
+    let(:initial_stage) { false }
+    let(:command_result) { { "exit" => 0 } }
+
+    before do
+        allow(Yast::Stage).to receive(:initial).and_return(initial_stage)
+
+        allow(Yast::WFM).to receive(:Execute).and_return(command_result)
+        allow(Yast::SCR).to receive(:Execute).and_return(command_result)
+        allow(Yast::SCR).to receive(:Write)
+        allow(subject).to receive(:valid_language?).with(language).and_return(true)
+
+        subject.Set(language)
+        subject.SetDefault
+    end
+
+    it "writes .sysconfig.language.INSTALLED_LANGUAGES" do
+      expect(Yast::SCR).to receive(:Write).with(Yast.path(".sysconfig.language.INSTALLED_LANGUAGES"), anything)
+
+      subject.Save
+    end
+
+    it "cleans .sysconfig.language in sysconfig" do
+      expect(Yast::SCR).to receive(:Write).with(Yast.path(".sysconfig.language"), nil)
+
+      subject.Save
+    end
+
+    context "when language is zh_HK" do
+      let(:language) { "zh_HK" }
+
+      it "sets LC_MESSAGES to zh_TW" do
+        expect(Yast::SCR).to receive(:Execute).with(anything, /.*localectl set-locale.*LC_MESSAGES.*zh_TW.*/)
+
+        subject.Save
+      end
+    end
+
+    context "when LC_MESAGES is zh_TW" do
+      before do
+        allow(subject).to receive(:@localed_conf).and_return({ "LC_MESSAGES" => "zh_TW" })
+      end
+
+      context "and language is not zh_HK" do
+        it "cleans LC_MESSAGES" do
+          expect(Yast::SCR).to_not receive(:Execute).with(anything, /.*localectl.*LC_MESSAGES.*zh_TW.*/)
```

Comment:
you test that it is not called, but it looks different then cleaning it for me. Does it really write empty string there?

Reply:
If it is cleaned (a.k.a [deleted]()), what I expect if that the 

> /usr/sbin/localectl set-locale

command does not receive the `LC_MESSAGE=zh_TW` as an argument. 

Is that approach wrong?